3_visualize.R

```diff
@@ -16,28 +30,105 @@ p3_targets <- list(
                     width = 10, height = 100, dpi = 300, limitsize = FALSE),
              format = "file" ),
 
-# set proj
+  # set projection
   tar_target(p2_proj,
              '+proj=aea +lat_0=23 +lon_0=-96 +lat_1=29.5 +lat_2=45.5 +x_0=0 +y_0=0 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs +type=crs'
              ),
 
-# transform to sf by filtering metadata to the unique StaID's for states that have the 2000 most severe droughts
+  # transform to sf by filtering metadata to the unique StaID's for states that have the 2000 most severe droughts
   tar_target(p2_metadata_sf,
              sf::st_as_sf(p2_metadata |> 
                             filter(StaID %in% unique(p2_2000_severe_2pct_droughts$StaID)), 
                           coords = c("LNG_GAGE", "LAT_GAGE"), crs = 4269) |> 
                st_transform(p2_proj)),
 
-# obtain all US wide data 
+  # obtain all US wide data 
   tar_target(p2_states,
              spData::us_states |> st_transform(p2_proj) |> ms_simplify(keep = 0.5)),
+  
 
-# plotting inset with stations
+  # plotting inset with stations
   tar_target(p3_inset_stations_map_png,
            plot_inset(station_data = p2_metadata_sf,
                       station = TRUE,
                       us_data = p2_states,
+                      regions = FALSE,
+                      region_sf = NA,
+                      focal_stations = FALSE,
+                      focal_station_data = p2_metadata_sf,
                       file_png = 'src/assets/images/states_stations_inset.png',
-                      width = 16, height = 9),
-           format = "file")
+                      width = 16, height = 9,
+                      color_scheme = p3_colors),
+           format = "file"),
+  
+  # create CASC-level violins
+  tar_target(p3_drought_violin_vertical_byCASC,
+             event_violin_vertical(drought_data = p2_expanded_2000_2pct_droughts_byCASC,
+                                   major_drought_periods = p2_major_droughts_expanded,
+                                   color_scheme = p3_colors),
+             pattern = map(p2_expanded_2000_2pct_droughts_byCASC),
+             format = "file"),
+
+  
+  # plotting inset with stations and regions
+  tar_target(p3_inset_stations_map_byCASC_png,
+             plot_inset(station_data = p2_metadata_sf |> 
+                          filter(STATE %in% p2_expanded_2000_2pct_droughts_byCASC$STATE),
+                        station = TRUE,
+                        us_data = p2_states,
+                        regions = TRUE,
+                        region_sf = p2_states |>
```

Comment:
SC: I think it wouldn't hurt to add a comment for this target explain that `region_sf` is built for each region from `p2_states` - spatially grouping states w/i each CASC and then passing only the geometry for the current CASC  to `plot_inset`

Reply:
I extracted this to a new target `p3_CASCs_sf` to make it clearer what is happening here and make it easier to reuse across 2+ targets